src/kixi/hecuba/api_helpers.clj

```diff
@@ -80,25 +80,57 @@
 (defn format-epoch-timestamp [epoch-timestamp]
   (c/to-string (* (read-string epoch-timestamp) 1000)))
 
+;; Here electricity files only contain electricity data
+;; but gas files also contain temperature and should
+;; be handled differently.
+(defn contains-temperature? [measurements]
+  (contains? (first measurements) :temperature))
+
+(defn format-map
+  "For a map with one measurement rename keys
+  and select the desired values."
+  [m m-key sensor-type]
+  (-> m
+      (clojure.set/rename-keys 
+       {:device_timestamp :timestamp
+        m-key :value})
+      (assoc :type sensor-type)
+      (select-keys [:timestamp :value :type])))
+
 (defn format-measurements
   "Input is a vector of maps with measurements and property 
   info to format into embed measurements."
-  [measurements]
-  (->> measurements
-       (map (fn [m]
-              (-> (clojure.set/rename-keys 
-                   {:device_timestamp :timestamp
-                    :energy :value})
-                  (assoc :type "electricityConsumption"))))
-       (mapv (fn [m] (update m :timestamp format-epoch-timestamp)))
-       (assoc {} :measurements)
-       json/write-str))
+  ([measurements key-type sensor-type]
+   (->> measurements
+        (map (fn [m]
+               (format-map m key-type sensor-type)))
+        (mapv (fn [m] (update m :timestamp format-epoch-timestamp)))
+        (assoc {} :measurements)
+        json/write-str)))
+
+(defn decide-upload
+  "Handle correctly epending on the type of the input data."
+  [entity-id device-id measurements base-url username password]
+  (if (contains-temperature? measurements)
+    (do (upload-measurements
+         entity-id device-id
+         (format-measurements measurements :energy "gasConsumption")
+         base-url username password)
```

Comment:
I can see `decide-upload` needs to be simplified. I think I've never used `partial` so I guess I need to experiment with it to understand how it can help me here.